### ISSUE TITLE
[clsim] 100 char unix socket limitation (Trac #2422)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2414]
Clsim server uses a ZMQ unix socket to communicate.  Unix sockets are limited to roughly 100 characters of absolute path length by the kernel.  Some sites are exceeding this length (specifically, UMD).

A workaround is needed for deeper directories.  Maybe UDP/TCP on localhost?

Migrated from https://code.icecube.wisc.edu/ticket/2422
```json
{
    "status": "closed", 
    "changetime": "2020-06-24T12:32:17", 
    "description": "Clsim server uses a ZMQ unix socket to communicate.  Unix sockets are limited to roughly 100 characters of absolute path length by the kernel.  Some sites are exceeding this length (specifically, UMD).\n\nA workaround is needed for deeper directories.  Maybe UDP/TCP on localhost?", 
    "reporter": "msilva", 
    "cc": "olivas", 
    "resolution": "fixed", 
    "_ts": "1593001937450890", 
    "component": "combo simulation", 
    "summary": "[clsim] 100 char unix socket limitation", 
    "priority": "major", 
    "keywords": "zmq", 
    "time": "2020-04-07T19:40:25", 
    "milestone": "Autumnal Equinox 2020", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20200407T21:12:52: jvansanten commented:

> Confirming that this is a path length issue. Reproduce with e.g.
> 
> ```text
> TMPDIR=/tmp/tmp.lmjaQmRtAx/45d93768-7913-11ea-bf58-90b11c480a2d-45d9d6c8-7913-11ea-8e59-90b11c480a2d-45da6ebc-7913-11ea-bbc3-90b11c480a2d clsim/resources/scripts/benchmark.py -n 1 --energy 1e-3
> terminate called after throwing an instance of 'zmq::error_t'
>   what():  File name too long
> [1]    41576 abort      TMPDIR= clsim/resources/scripts/benchmark.py -n 1 --energy 1e-3
> ```
> if TMPDIR is not set, works as expected.

**Comment by icecube-trac** [comment]
Trac update at 20200408T09:54:16: 
* jvansanten changed resolution from "" to "fixed"
* jvansanten changed status from "new" to "closed"
* jvansanten commented:

> In changeset 179700:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="179700"
> Bind clsim server to random local TCP port
> 
> This works around an issue where mktemp() uses a prefix that is too long
> to be a valid socket name on some kernels. Fixes #2414.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20200624T12:32:17: 
* olivas changed milestone from "Summer Solstice 2020" to "Autumnal Equinox 2020"
* olivas commented:

> Milestone Summer Solstice 2020 deleted
